jupyter_notebooks/Analyzing public and hybrid cloud networks .ipynb

```diff
@@ -0,0 +1,567 @@
+    "    \"\"\"\n",
+    "    show(\"Flow: {}\".format(trace_answer_frame.iloc[0]['Flow']))\n",
+    "    show(trace_answer_frame.iloc[0]['Traces'][0])\n",
+    "    \n",
+    "def is_reachable(start_location, end_location, headers=None):\n",
+    "    \"\"\"\n",
+    "    Checks if the start_location can reach the end_location using specified packet headers.\n",
+    "    \n",
+    "    All possible headers are considered if headers is None. \n",
+    "    \"\"\"\n",
+    "    ans = bfq.reachability(pathConstraints=PathConstraints(startLocation=start_location, \n",
+    "                                                           endLocation=end_location),\n",
+    "                          headers=headers).answer()\n",
+    "    return len(ans.frame()) > 0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Initializing the Network and Snapshot\n",
+    "\n",
+    "SNAPSHOT_PATH below can be updated to point to a custom snapshot directory. See [instructions](https://github.com/batfish/batfish/wiki/Packaging-snapshots-for-analysis) for how to package data for analysis. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'snapshot'"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Initialize a network and snapshot\n",
+    "NETWORK_NAME = \"hybrid-cloud\"\n",
+    "SNAPSHOT_NAME = \"snapshot\"\n",
+    "\n",
+    "SNAPSHOT_PATH = \"networks/hybrid-cloud\"\n",
+    "\n",
+    "bf_set_network(NETWORK_NAME)\n",
+    "bf_init_snapshot(SNAPSHOT_PATH, name=SNAPSHOT_NAME, overwrite=True, extra_args={ \"parsereuse\" : False})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The network snapshot that we just initialized is illustrated below. It has a datacenter network with the standard leaf-spine design on the left. Though not strictly necessary, we have included a host `srv-101` in this network to enable end-to-end analysis. The exit gateway of the datacenter connects to an Internet service provider that we call `isp_dc`.\n",
+    "\n",
+    "The AWS network is shown on the right. It is spread across two regions, `us-east-2` and `us-west-2`. Each region has two VPCs, one of which is meant to host Internet-facing services and the other is meant to host only private services. Subnets in the public-facing VPCs use an Internet gateway to send and receive traffic outside of AWS. The two VPCs in a region peer via a transit gateway. Each VPC has two subnets, and we have some instances running as well. \n",
+    "\n",
+    "The physical network connects to the AWS network by having the exitgw establish IPSec tunnels, shown in pink, to the two transit gateways. BGP sessions running atop these tunnels to make endpoints aware of prefixes on the other side.\n",
```

Comment:
`exitgw` should be in code-font.

replace "running" with "run" below...
"BGP sessions run atop these tunnels to make endpoints aware of prefixes..."